parser: allow string literals for region names

otan opened the pull request:
Release note (sql change): Allow string literals for region names in DDL syntax instead of just quoted syntax.

Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/97021)
<!-- Reviewable:end -->

otan committed:
parser: allow string literals for region names

Release note (sql change): Allow string literals for region names in DDL
syntax instead of just quoted syntax.

knz commented:
IMHO we should allow _only_ string literals and disallow identifiers. This would allow us to (later) use correlated names to specify regions in complex queries.

otan commented:
i agree. i think originally we didn't want "quotes" to wrap region names, e.g. `REGIONS a-region, b-region` but that didn't work out. we can look at deprecating the old way later perhaps.

bors r=knz
